main/src/modules/attentionRelated/ears/src/ears.cpp

```diff
@@ -219,28 +277,98 @@ bool ears::updateModule() {
             sObject = "";
         } else {
             yError() << "[ears] Unknown predicate";
-
+            if (followPlans){
+                    Bottle &bGoal = goal_need_port.prepare();
+                    bGoal.clear();
+                    bGoal.addInt(1);
+                    goal_need_port.write();
+                }
+            return true;
         }
-
-        Bottle &bToTarget = portTarget.prepare();
-        bToTarget.clear();
-        bToTarget.addString(sAction);
-        bToTarget.addString(sObjectType);
-        bToTarget.addString(sObject);
-        portTarget.write();
-
+        //send rpc data to goalManager
+        if (followPlans){
+            Bottle &bToTarget = port_planner.prepare();
+            bToTarget.clear();
+            Bottle bAux;
+            bAux.clear();
+            Bottle bAux2;
+            bAux2.clear();
+            bToTarget.addString("new");
+            bAux.addString(sAction);
+            bAux2.addString(sObjectType);
+            bAux2.addString(sObject);
+            bAux.addList()=bAux2;
+            bToTarget.addList()=bAux;
+            port_planner.write();
+            yDebug() << "Sending " + bToTarget.toString();
+        }else{
+            Bottle &bToTarget = portTarget.prepare();
+            bToTarget.clear();
+            bToTarget.addString(sAction);
+            bToTarget.addString(sObjectType);
+            bToTarget.addString(sObject);
+            portTarget.write();
+
+            Bottle bCondition;
+            bCondition.addString(sCommand);
+            bCondition.addString(sAction);
+            bCondition.addString(sObjectType);
+            bCondition.addString(sObject);
+
+            portToBehavior.write(bCondition);
```

Comment:
Ideally, `ears` and `behaviorManager` will both check the variable `onPlannerMode` in their config files. If it is `false`, it will behave as per the year 2 demo. If it is `true`, then the planner module will take effect.